rust/operator-binary/src/crd/mod.rs

```diff
@@ -435,27 +505,31 @@ pub struct AirflowOpaConfig {
     pub cache: UserInformationCache,
 }
 
-// TODO: Temporary solution until listener-operator is finished
 #[derive(Clone, Debug, Default, Display, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "PascalCase")]
-pub enum CurrentlySupportedListenerClasses {
+pub enum SupportedListenerClasses {
     #[default]
     #[serde(rename = "cluster-internal")]
+    #[strum(serialize = "cluster-internal")]
     ClusterInternal,
 
     #[serde(rename = "external-unstable")]
+    #[strum(serialize = "external-unstable")]
     ExternalUnstable,
 
     #[serde(rename = "external-stable")]
+    #[strum(serialize = "external-stable")]
     ExternalStable,
 }
 
-impl CurrentlySupportedListenerClasses {
-    pub fn k8s_service_type(&self) -> String {
+impl Atomic for SupportedListenerClasses {}
+
+impl SupportedListenerClasses {
+    pub fn discoverable(&self) -> bool {
         match self {
-            CurrentlySupportedListenerClasses::ClusterInternal => "ClusterIP".to_string(),
-            CurrentlySupportedListenerClasses::ExternalUnstable => "NodePort".to_string(),
-            CurrentlySupportedListenerClasses::ExternalStable => "LoadBalancer".to_string(),
+            SupportedListenerClasses::ClusterInternal => false,
+            SupportedListenerClasses::ExternalUnstable => true,
+            SupportedListenerClasses::ExternalStable => true,
         }
     }
 }
```

Comment:
Now that we're using listener-operator, the listener class name should just be an opaque string.

Reply:
Looking at the implementation for kafka-operator, a PVC is created for the bootstrap listener even when it is not defined by the user (defaulting to cluster-internal). I guess there is no easy way around that, other than looking up the listener class and checking its type?

Reply:
kafka-operator behaves correctly there; even when running cluster-internally we want the Listener -> Service to be the ingress point that we expose.